File: Flink_WuShengran/src/main/scala/com/atguigu/wc/WordCount.scala
package com.atguigu.wc

import org.apache.flink.api.scala._

/**
 * Created by John.Ma on 2020/9/10 0010 22:19
 * 1.批处理的word count
 */
object WordCount {
    def main(args: Array[String]): Unit = {
        //创建一个批处理的执行环境
        val env: ExecutionEnvironment = ExecutionEnvironment.getExecutionEnvironment

        //从文件中读取数据
        val inputPath = "D:\\fileImportant\\Learn_projects\\learn\\Flink_WuShengran\\src\\main\\resources\\hello.txt"
        val inputDataSet = env.readTextFile(inputPath)

        //对数据进行转换处理统计,先分词,再按照word进行分组,最后进行聚合统计
        val resultDataSet: DataSet[(String, Int)] = inputDataSet
            .flatMap(_.split(" "))
            .map((_, 1))
            .groupBy(0) //以第一个元组作为key进行分组
            .sum(1)

        //打印输出
        resultDataSet.print()

    }
}

File: Flink_WuShengranJava/src/main/java/com/atguigu/wc/WordCount.java
package com.atguigu.wc;

import org.apache.flink.api.common.functions.RichFlatMapFunction;
import org.apache.flink.api.java.DataSet;
import org.apache.flink.api.java.ExecutionEnvironment;
import org.apache.flink.api.java.operators.DataSource;
import org.apache.flink.api.java.operators.FlatMapOperator;
import org.apache.flink.api.java.tuple.Tuple2;
import org.apache.flink.util.Collector;

import java.io.InputStream;

/**
 * 批处理word count
 */
public class WordCount {
    public static void main(String[] args) throws Exception {
        //1.创建执行环境
        ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();

        // 从文件中读取器数据
        DataSource<String> inputDataSet = env.readTextFile("D:\\fileImportant\\Learn_projects\\learn\\Flink_WuShengranJava\\src\\main\\resources\\hello.txt");

        // 对数据集进行处理,按空格分词展开,转换成(word,1L)
        DataSet<Tuple2<String,Long>> resultSet = inputDataSet
                .flatMap(new RichFlatMapFunction<String, Tuple2<String, Long>>() {
                    @Override
                    public void flatMap(String s, Collector<Tuple2<String, Long>> collector) throws Exception {
                        String[] arr = s.split(" ");
                        for (String s1 : arr) {
                            collector.collect(Tuple2.of(s1, 1L));
                        }
                    }
                })
                .groupBy(0)
                .sum(1);

        resultSet.print();

    }
}
